test/core/utils/flattened-tree.js

```diff
@@ -86,11 +89,20 @@ describe('axe.utils.getFlattenedTree', function() {
     assert(tree[0].parent === null);
   });
 
+  it('creates virtual nodes in the correct order', function() {
+    fixture.innerHTML = '<p><b><i></i></b><p><u><s></s></u>';
+
+    var vNode = axe.utils.getFlattenedTree(fixture)[0];
+    assert.equal(vNode.nodeIndex, 0);
+    assert.equal(vNode.children[0].nodeIndex, 1);
+    assert.equal(vNode.children[0].children[0].nodeIndex, 2);
+    assert.equal(vNode.children[0].children[0].children[0].nodeIndex, 3);
+    assert.equal(vNode.children[1].nodeIndex, 4);
```

Comment:
This test would be more resilient if the code tested what the element was as well as testing node indexes. This line expects to be testing the `u` element but is instead testing the extraneous `p` element that was created.